### ISSUE TITLE
DetachDomain destroys all remote ssh mux panes.

### ISSUE THREAD

**Issue by loops** [wez/wezterm#1993]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### WezTerm version

wezterm 20220514-175933-64b6e63f

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

When the Gui client is connected to a single ssh Domain everything works as expected.  The DetachDomain action disconnects from the remote server, and leaves all of the panes there running, they can be reconnected to in the future.

However, if the Gui client is connected to an additional domain, then that same DetachDomain action kills all the remote panes instead of leaving them running.

### To Reproduce

//  With the local wezterm configuration shown below, and with nothing yet running on the remote ssh server:

1. wezterm start   (creates single tab, with local shell)
2. shift-ctrl-j   (creates a new ssh tab, with "wezterm cli --prefer-mux proxy", and "wezterm-mux-server" running on remote ssh host)
3. shift-ctrl-l  and then ctrl-d  (switch back to the tab with the local shell, and exit that shell, closing the local domain)
4. shift-ctrl-k  (detach the remote ssh server)

If you then use a different program to ssh into the remote ssh server, you'll see that  "wezterm-mux-server" is still running, and issuing the command "wezterm cli list" lists a single pane as still running.

However, if you start all over (with nothing yet running on the remote ssh server) and repeat the same steps above, except skipping step 3.  No pane will remain running afterward.   This seems to be because the local domain is still active when the detach action is issued.  That step 3, closes the local domain, and allows the detach to work as expected.

This example only started a single remote pane on the ssh mux server, however if there are 2, 3 or any number more, they are all shut down.

### Configuration

```lua
local wezterm = require 'wezterm';

return {
  font = wezterm.font("DejaVuSansMono Nerd Font Mono"),
  font_size = 16.0,
  enable_wayland = true,
  enable_tab_bar = false,
  window_background_opacity = 0.0,
  window_padding = {
    left = 3,
    right = 2,
    top = 3,
    bottom = 2,
  },
  exit_behavior = "Close",
  enable_csi_u_key_encoding = true,
  keys = {
    {key="l", mods="SHIFT|CTRL", action=wezterm.action{ActivateTabRelative=1}},
    {key="h", mods="SHIFT|CTRL", action=wezterm.action{ActivateTabRelative=-1}},
    {key="j", mods="CTRL|SHIFT", action=wezterm.action{AttachDomain="vul"}},
    {key="k", mods="CTRL|SHIFT", action=wezterm.action{DetachDomain={DomainName="vul"}}},
  },
  ssh_domains = {
    {
      name = "vul",
      remote_address = "vultr",
      username = "xxx",
      connect_automatically = false,
      remote_wezterm_path = "/home/xxx/bin/wezterm",
    },
  },
  mouse_bindings = {
    { event={Down={streak=3, button="Left"}},
      action={SelectTextAtMouseCursor="SemanticZone"},
      mods="NONE"
    },
  },
}
```

### Expected Behavior

Would expect that the detach action would always leave the remote panes running and able to be reattached in the future.

### Logs

```
> wezterm start
18:19:37.871  INFO   wezterm_gui::termwindow > OpenGL initialized! AMD RENOIR (LLVM 13.0.1, DRM 3.44, 5.17.6-200.fc35.x86_64) 4.6 (Compatibility Profile) Mesa 22.0.1 is_context_loss_possible=false wezterm version: 20220514-175933-64b6e63f
18:19:44.095  ERROR  wezterm_client::client  > going to run /home/xxx/bin/wezterm cli --prefer-mux proxy
18:19:44.963  ERROR  wezterm_client::client  > ssh stderr: 18:19:44.947  WARN   wezterm_client::client > While connecting to Socket("/run/user/1000/wezterm/sock"): connecting to /run/user/1000/wezterm/sock.  Will try spawning the server.
18:19:44.991  ERROR  wezterm_client::client  > ssh stderr: 18:19:44.950  WARN   wezterm_client::client > Running: "/home/xxx/bin/wezterm-mux-server" "--daemonize"
18:19:49.179  ERROR  wezterm_client::domain  > detached domain 2
18:19:49.179  ERROR  mux                     > domain detached panes: [2]
18:19:49.415  ERROR  wezterm_client::domain  > detached domain 2
18:19:49.415  ERROR  mux                     > domain detached panes: []
```

### Anything else?

When there are 2 domains active, issuing the "DetachDomain" action kills the remote ssh mux panes.  However, if instead of issuing that action, the local GUI is just killed (Linux kill command), the remote panes continue to exist and can be attached to in the future.   So it seems to be explicitly the "DetachDomain" command at fault.

**Comment by wez** [comment]
This should be fixed now in `main`.

It typically takes about an hour before fixes are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out.

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them.

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Comment by loops** [comment]
Built from source and all is looking good.  Works as advertised.   Thank you Wez!

**Comment by loops** [comment]
Sorry, but I just noticed, that when I disconnect from the remote domain now, CPU usage goes to 100%  for the wezterm-gui process.   Does this happen for anyone else?    What could I do to help debug this?

**Comment by wez** [comment]
c8c3b8378a5f7b25a3bd56bf6e25d8b8d11d3efc should resolve that!

**Comment by loops** [comment]
> [c8c3b83](https://github.com/wez/wezterm/commit/c8c3b8378a5f7b25a3bd56bf6e25d8b8d11d3efc) should resolve that!

Thanks again Wez, that seems to do the trick!

As an aside, moving to the latest HEAD forced a config modification, since  the SwitchToWorkspace action now requires the spawn parameter to be supplied explicitly. Not a problem to fix up, but thought I'd mention it, in case that wasn't intended.

**Comment by wez** [comment]
> As an aside, moving to the latest HEAD forced a config modification, since the SwitchToWorkspace action now requires the spawn parameter to be supplied explicitly. Not a problem to fix up, but thought I'd mention it, in case that wasn't intended.

Can you share the config snippet that didn't work out and the error that you saw?

**Comment by loops** [comment]
> Can you share the config snippet that didn't work out and the error that you saw?

callback error
stack traceback:
        [C]: in function 'wezterm.action'
        [string "/home/sean/.config/wezterm/wezterm.lua"]:62: in main chunk
caused by: error converting Lua table to KeyAssignment (Error processing
SwitchToWorkspace::spawn: missing field)


Config line 62 was:

   "{key="y", mods="CTRL|SHIFT", action=wezterm.action{SwitchToWorkspace={ name = "default"}}},"

And changing it to:

{key="y", mods="CTRL|SHIFT", action=wezterm.action{SwitchToWorkspace={ name = "default", spawn={domain={DomainName="default"}}}}},

Made the error go away.  The desire was to define a key to switch back to the workspace that is created implicitly when none is named on startup.

**Comment by wez** [comment]
Thanks; should be fixed in 94039c473bebbf35d4994863965ebd58c3647218

**Comment by loops** [comment]
> Thanks; should be fixed in [94039c4](https://github.com/wez/wezterm/commit/94039c473bebbf35d4994863965ebd58c3647218)

That worked a treat.

Not sure on your preference, but I think you've earned the right to close this issue :-)   

Thanks again.
